doxia-core/src/main/java/org/apache/maven/doxia/parser/Parser.java

```diff
@@ -83,4 +85,34 @@ public interface Parser {
      * @return <code>true</code> (default value) if comment Doxia events are emitted
      */
     boolean isEmitComments();
+
+    /**
+     * Registers a given {@link SinkWrapperFactory} with the parser used in subsequent calls of {@code parse(...)}
+     * @param factory the factory to create the sink wrapper
+     * @since 2.0.0
+     */
+    void registerSinkWrapperFactory(SinkWrapperFactory factory);
+
+    /**
+     *
+     * @return all sink wrapper factories in the correct order (both registered automatically and manually)
+     * @since 2.0.0
+     */
+    Collection<SinkWrapperFactory> getSinkWrapperFactories();
+
+    /**
+     * Determines whether to automatically generate anchors for each section found by {@link IndexingSink} or not.
+     * By default no anchors are generated.
+     *
+     * @param emitAnchors {@code true} to emit anchors otherwise {@code false} (the default)
+     * @since 2.0.0
+     */
+    void setEmitAnchors(boolean emitAnchors);
```

Comment:
This should be more specific: `setEmitSectionTitleAnchors`

Reply:
It is potentially for all index entries (although IndexingSink currently only generates ones for sections, but there are some todos for figures, tables, ...). For sure it is not specific to to ones having a title....

Reply:
Maybe `setEmitAnchorsForIndexEntries(...)`?

Reply:
You mean everything would could potentially have a number group? Then maybe `setEmitAnchorsForIndexableEntries()` and docs will carry only the information that it applies to section titles for the moment?